src/preamble_minimal.js

```diff
@@ -77,6 +77,22 @@ function updateMemoryViews() {
   HEAP64 = new BigInt64Array(b);
   HEAPU64 = new BigUint64Array(b);
 #endif
+
+#if MODULARIZE && EXPORT_ALL
+  // Export the heap to be used outside of the WASM module.
+  Module['HEAP8'] = HEAP8;
+  Module['HEAP16'] = HEAP16;
+  Module['HEAP32'] = HEAP32;
+  Module['HEAPU8'] = HEAPU8;
+  Module['HEAPU16'] = HEAPU16;
+  Module['HEAPU32'] = HEAPU32;
+  Module['HEAPF32'] = HEAPF32;
+  Module['HEAPF64'] = HEAPF64;
+#if WASM_BIGINT
+  Module['HEAP64'] = HEAP64;
+  Module['HEAPU64'] = HEAPU64;
+#endif
+#endif
```

Comment:
With a little bit of magic with the `{{{ }}}` syntax, the above can be unified into

```diff
diff --git a/src/preamble_minimal.js b/src/preamble_minimal.js
index 2f32865d9..56e96df31 100644
--- a/src/preamble_minimal.js
+++ b/src/preamble_minimal.js
@@ -62,20 +62,28 @@ function updateMemoryViews() {
 #if ASSERTIONS && SHARED_MEMORY
   assert(b instanceof SharedArrayBuffer, 'requested a shared WebAssembly.Memory but the returned buffer is not a SharedArrayBuffer, indicating that while the browser has SharedArrayBuffer it does not have WebAssembly threads support - you may need to set a flag');
 #endif
+
+{{{ (function() {
+  global.assignToModuleIfExportAll = function(x) {
+    return MODULARIZE && EXPORT_ALL ? `Module['${x}'] =` : '';
+  };
+  return null;
+})(); }}}
+
 #if SUPPORT_BIG_ENDIAN
-  HEAP_DATA_VIEW = new DataView(b);
-#endif
-  HEAP8 = new Int8Array(b);
-  HEAP16 = new Int16Array(b);
-  HEAP32 = new Int32Array(b);
-  HEAPU8 = new Uint8Array(b);
-  HEAPU16 = new Uint16Array(b);
-  HEAPU32 = new Uint32Array(b);
-  HEAPF32 = new Float32Array(b);
-  HEAPF64 = new Float64Array(b);
+  {{{ assignToModuleIfExportAll('HEAP_DATA_VIEW') }}} HEAP_DATA_VIEW = new DataView(b);
+#endif
+  {{{ assignToModuleIfExportAll('HEAP8') }}} HEAP8 = new Int8Array(b);
+  {{{ assignToModuleIfExportAll('HEAP16') }}} HEAP16 = new Int16Array(b);
+  {{{ assignToModuleIfExportAll('HEAP32') }}} HEAP32 = new Int32Array(b);
+  {{{ assignToModuleIfExportAll('HEAPU8') }}} HEAPU8 = new Uint8Array(b);
+  {{{ assignToModuleIfExportAll('HEAPU16') }}} HEAPU16 = new Uint16Array(b);
+  {{{ assignToModuleIfExportAll('HEAPU32') }}} HEAPU32 = new Uint32Array(b);
+  {{{ assignToModuleIfExportAll('HEAPF32') }}} HEAPF32 = new Float32Array(b);
+  {{{ assignToModuleIfExportAll('HEAPF64') }}} HEAPF64 = new Float64Array(b);
 #if WASM_BIGINT
-  HEAP64 = new BigInt64Array(b);
-  HEAPU64 = new BigUint64Array(b);
+  {{{ assignToModuleIfExportAll('HEAP64') }}} HEAP64 = new BigInt64Array(b);
+  {{{ assignToModuleIfExportAll('HEAPU64') }}} HEAPU64 = new BigUint64Array(b);
 #endif
 }
```
to improve code size.

Reply:
> The HEAP is going to be exported only using EXPORT_ALL. So, someone manually selecting what to export will not have it. Not sure what would be the best way to fix that.

So far in MINIMAL_RUNTIME there does not exist any kind of mechanism to export out to `Module`, so this `EXPORT_ALL` flag will be the first mechanism to enable that for `MINIMAL_RUNTIME`. But if we do build such a method in the future, then the above type of `assignToModuleIfExportAll()` function would be able to look into an exports array per each symbol to see which assignments to outside world it would want to do.

Reply:
I like this idea, done :)